/*
 * Genome Nexus API
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * OpenAPI spec version: 2.0
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


package org.genome_nexus.client;

import java.util.Objects;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

/**
 * Vues
 */
@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaClientCodegen", date = "2023-06-15T10:43:13.566-05:00")
public class Vues {
  @SerializedName("comment")
  private String comment = null;

  @SerializedName("defaultEffect")
  private String defaultEffect = null;

  @SerializedName("genomicLocation")
  private String genomicLocation = null;

  @SerializedName("genomicLocationDescription")
  private String genomicLocationDescription = null;

  @SerializedName("hugoGeneSymbol")
  private String hugoGeneSymbol = null;

  @SerializedName("pubmedIds")
  private List<Integer> pubmedIds = null;

  @SerializedName("referenceText")
  private String referenceText = null;

  @SerializedName("revisedProteinEffect")
  private String revisedProteinEffect = null;

  @SerializedName("transcriptId")
  private String transcriptId = null;

  @SerializedName("variant")
  private String variant = null;

  @SerializedName("variantClassification")
  private String variantClassification = null;

  public Vues comment(String comment) {
    this.comment = comment;
    return this;
  }

   /**
   * Get comment
   * @return comment
  **/
  @ApiModelProperty(value = "")
  public String getComment() {
    return comment;
  }

  public void setComment(String comment) {
    this.comment = comment;
  }

  public Vues defaultEffect(String defaultEffect) {
    this.defaultEffect = defaultEffect;
    return this;
  }

   /**
   * Get defaultEffect
   * @return defaultEffect
  **/
  @ApiModelProperty(value = "")
  public String getDefaultEffect() {
    return defaultEffect;
  }

  public void setDefaultEffect(String defaultEffect) {
    this.defaultEffect = defaultEffect;
  }

  public Vues genomicLocation(String genomicLocation) {
    this.genomicLocation = genomicLocation;
    return this;
  }

   /**
   * Get genomicLocation
   * @return genomicLocation
  **/
  @ApiModelProperty(value = "")
  public String getGenomicLocation() {
    return genomicLocation;
  }

  public void setGenomicLocation(String genomicLocation) {
    this.genomicLocation = genomicLocation;
  }

  public Vues genomicLocationDescription(String genomicLocationDescription) {
    this.genomicLocationDescription = genomicLocationDescription;
    return this;
  }

   /**
   * Get genomicLocationDescription
   * @return genomicLocationDescription
  **/
  @ApiModelProperty(value = "")
  public String getGenomicLocationDescription() {
    return genomicLocationDescription;
  }

  public void setGenomicLocationDescription(String genomicLocationDescription) {
    this.genomicLocationDescription = genomicLocationDescription;
  }

  public Vues hugoGeneSymbol(String hugoGeneSymbol) {
    this.hugoGeneSymbol = hugoGeneSymbol;
    return this;
  }

   /**
   * Get hugoGeneSymbol
   * @return hugoGeneSymbol
  **/
  @ApiModelProperty(value = "")
  public String getHugoGeneSymbol() {
    return hugoGeneSymbol;
  }

  public void setHugoGeneSymbol(String hugoGeneSymbol) {
    this.hugoGeneSymbol = hugoGeneSymbol;
  }

  public Vues pubmedIds(List<Integer> pubmedIds) {
    this.pubmedIds = pubmedIds;
    return this;
  }

  public Vues addPubmedIdsItem(Integer pubmedIdsItem) {
    if (this.pubmedIds == null) {
      this.pubmedIds = new ArrayList<Integer>();
    }
    this.pubmedIds.add(pubmedIdsItem);
    return this;
  }

   /**
   * Get pubmedIds
   * @return pubmedIds
  **/
  @ApiModelProperty(value = "")
  public List<Integer> getPubmedIds() {
    return pubmedIds;
  }

  public void setPubmedIds(List<Integer> pubmedIds) {
    this.pubmedIds = pubmedIds;
  }

  public Vues referenceText(String referenceText) {
    this.referenceText = referenceText;
    return this;
  }

   /**
   * Get referenceText
   * @return referenceText
  **/
  @ApiModelProperty(value = "")
  public String getReferenceText() {
    return referenceText;
  }

  public void setReferenceText(String referenceText) {
    this.referenceText = referenceText;
  }

  public Vues revisedProteinEffect(String revisedProteinEffect) {
    this.revisedProteinEffect = revisedProteinEffect;
    return this;
  }

   /**
   * Get revisedProteinEffect
   * @return revisedProteinEffect
  **/
  @ApiModelProperty(value = "")
  public String getRevisedProteinEffect() {
    return revisedProteinEffect;
  }

  public void setRevisedProteinEffect(String revisedProteinEffect) {
    this.revisedProteinEffect = revisedProteinEffect;
  }

  public Vues transcriptId(String transcriptId) {
    this.transcriptId = transcriptId;
    return this;
  }

   /**
   * Get transcriptId
   * @return transcriptId
  **/
  @ApiModelProperty(value = "")
  public String getTranscriptId() {
    return transcriptId;
  }

  public void setTranscriptId(String transcriptId) {
    this.transcriptId = transcriptId;
  }

  public Vues variant(String variant) {
    this.variant = variant;
    return this;
  }

   /**
   * Get variant
   * @return variant
  **/
  @ApiModelProperty(value = "")
  public String getVariant() {
    return variant;
  }

  public void setVariant(String variant) {
    this.variant = variant;
  }

  public Vues variantClassification(String variantClassification) {
    this.variantClassification = variantClassification;
    return this;
  }

   /**
   * Get variantClassification
   * @return variantClassification
  **/
  @ApiModelProperty(value = "")
  public String getVariantClassification() {
    return variantClassification;
  }

  public void setVariantClassification(String variantClassification) {
    this.variantClassification = variantClassification;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Vues vues = (Vues) o;
    return Objects.equals(this.comment, vues.comment) &&
        Objects.equals(this.defaultEffect, vues.defaultEffect) &&
        Objects.equals(this.genomicLocation, vues.genomicLocation) &&
        Objects.equals(this.genomicLocationDescription, vues.genomicLocationDescription) &&
        Objects.equals(this.hugoGeneSymbol, vues.hugoGeneSymbol) &&
        Objects.equals(this.pubmedIds, vues.pubmedIds) &&
        Objects.equals(this.referenceText, vues.referenceText) &&
        Objects.equals(this.revisedProteinEffect, vues.revisedProteinEffect) &&
        Objects.equals(this.transcriptId, vues.transcriptId) &&
        Objects.equals(this.variant, vues.variant) &&
        Objects.equals(this.variantClassification, vues.variantClassification);
  }

  @Override
  public int hashCode() {
    return Objects.hash(comment, defaultEffect, genomicLocation, genomicLocationDescription, hugoGeneSymbol, pubmedIds, referenceText, revisedProteinEffect, transcriptId, variant, variantClassification);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Vues {\n");
    
    sb.append("    comment: ").append(toIndentedString(comment)).append("\n");
    sb.append("    defaultEffect: ").append(toIndentedString(defaultEffect)).append("\n");
    sb.append("    genomicLocation: ").append(toIndentedString(genomicLocation)).append("\n");
    sb.append("    genomicLocationDescription: ").append(toIndentedString(genomicLocationDescription)).append("\n");
    sb.append("    hugoGeneSymbol: ").append(toIndentedString(hugoGeneSymbol)).append("\n");
    sb.append("    pubmedIds: ").append(toIndentedString(pubmedIds)).append("\n");
    sb.append("    referenceText: ").append(toIndentedString(referenceText)).append("\n");
    sb.append("    revisedProteinEffect: ").append(toIndentedString(revisedProteinEffect)).append("\n");
    sb.append("    transcriptId: ").append(toIndentedString(transcriptId)).append("\n");
    sb.append("    variant: ").append(toIndentedString(variant)).append("\n");
    sb.append("    variantClassification: ").append(toIndentedString(variantClassification)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

