/*
 * Genome Nexus API
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package org.genome_nexus.client;

import java.util.Objects;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonArray;
import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;
import com.google.gson.TypeAdapterFactory;
import com.google.gson.reflect.TypeToken;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

import org.genome_nexus.JSON;

/**
 * EnsemblFilter
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", date = "2023-12-26T11:53:02.948968527-05:00[America/New_York]")
public class EnsemblFilter {
  public static final String SERIALIZED_NAME_GENE_IDS = "geneIds";
  @SerializedName(SERIALIZED_NAME_GENE_IDS)
  private List<String> geneIds;

  public static final String SERIALIZED_NAME_HUGO_SYMBOLS = "hugoSymbols";
  @SerializedName(SERIALIZED_NAME_HUGO_SYMBOLS)
  private List<String> hugoSymbols;

  public static final String SERIALIZED_NAME_PROTEIN_IDS = "proteinIds";
  @SerializedName(SERIALIZED_NAME_PROTEIN_IDS)
  private List<String> proteinIds;

  public static final String SERIALIZED_NAME_TRANSCRIPT_IDS = "transcriptIds";
  @SerializedName(SERIALIZED_NAME_TRANSCRIPT_IDS)
  private List<String> transcriptIds;

  public EnsemblFilter() {
  }

  public EnsemblFilter geneIds(List<String> geneIds) {
    this.geneIds = geneIds;
    return this;
  }

  public EnsemblFilter addGeneIdsItem(String geneIdsItem) {
    if (this.geneIds == null) {
      this.geneIds = new ArrayList<>();
    }
    this.geneIds.add(geneIdsItem);
    return this;
  }

   /**
   * List of Ensembl gene IDs. For example [\&quot;ENSG00000136999\&quot;, \&quot;ENSG00000272398\&quot;, \&quot;ENSG00000198695\&quot;]
   * @return geneIds
  **/
  @javax.annotation.Nullable
  public List<String> getGeneIds() {
    return geneIds;
  }

  public void setGeneIds(List<String> geneIds) {
    this.geneIds = geneIds;
  }


  public EnsemblFilter hugoSymbols(List<String> hugoSymbols) {
    this.hugoSymbols = hugoSymbols;
    return this;
  }

  public EnsemblFilter addHugoSymbolsItem(String hugoSymbolsItem) {
    if (this.hugoSymbols == null) {
      this.hugoSymbols = new ArrayList<>();
    }
    this.hugoSymbols.add(hugoSymbolsItem);
    return this;
  }

   /**
   * List of Hugo Symbols. For example [\&quot;TP53\&quot;, \&quot;PIK3CA\&quot;, \&quot;BRCA1\&quot;]
   * @return hugoSymbols
  **/
  @javax.annotation.Nullable
  public List<String> getHugoSymbols() {
    return hugoSymbols;
  }

  public void setHugoSymbols(List<String> hugoSymbols) {
    this.hugoSymbols = hugoSymbols;
  }


  public EnsemblFilter proteinIds(List<String> proteinIds) {
    this.proteinIds = proteinIds;
    return this;
  }

  public EnsemblFilter addProteinIdsItem(String proteinIdsItem) {
    if (this.proteinIds == null) {
      this.proteinIds = new ArrayList<>();
    }
    this.proteinIds.add(proteinIdsItem);
    return this;
  }

   /**
   * List of Ensembl protein IDs. For example [\&quot;ENSP00000439985\&quot;, \&quot;ENSP00000478460\&quot;, \&quot;ENSP00000346196\&quot;]
   * @return proteinIds
  **/
  @javax.annotation.Nullable
  public List<String> getProteinIds() {
    return proteinIds;
  }

  public void setProteinIds(List<String> proteinIds) {
    this.proteinIds = proteinIds;
  }


  public EnsemblFilter transcriptIds(List<String> transcriptIds) {
    this.transcriptIds = transcriptIds;
    return this;
  }

  public EnsemblFilter addTranscriptIdsItem(String transcriptIdsItem) {
    if (this.transcriptIds == null) {
      this.transcriptIds = new ArrayList<>();
    }
    this.transcriptIds.add(transcriptIdsItem);
    return this;
  }

   /**
   * List of Ensembl transcript IDs. For example [\&quot;ENST00000361390\&quot;, \&quot;ENST00000361453\&quot;, \&quot;ENST00000361624\&quot;]
   * @return transcriptIds
  **/
  @javax.annotation.Nullable
  public List<String> getTranscriptIds() {
    return transcriptIds;
  }

  public void setTranscriptIds(List<String> transcriptIds) {
    this.transcriptIds = transcriptIds;
  }



  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    EnsemblFilter ensemblFilter = (EnsemblFilter) o;
    return Objects.equals(this.geneIds, ensemblFilter.geneIds) &&
        Objects.equals(this.hugoSymbols, ensemblFilter.hugoSymbols) &&
        Objects.equals(this.proteinIds, ensemblFilter.proteinIds) &&
        Objects.equals(this.transcriptIds, ensemblFilter.transcriptIds);
  }

  @Override
  public int hashCode() {
    return Objects.hash(geneIds, hugoSymbols, proteinIds, transcriptIds);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class EnsemblFilter {\n");
    sb.append("    geneIds: ").append(toIndentedString(geneIds)).append("\n");
    sb.append("    hugoSymbols: ").append(toIndentedString(hugoSymbols)).append("\n");
    sb.append("    proteinIds: ").append(toIndentedString(proteinIds)).append("\n");
    sb.append("    transcriptIds: ").append(toIndentedString(transcriptIds)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }


  public static HashSet<String> openapiFields;
  public static HashSet<String> openapiRequiredFields;

  static {
    // a set of all properties/fields (JSON key names)
    openapiFields = new HashSet<String>();
    openapiFields.add("geneIds");
    openapiFields.add("hugoSymbols");
    openapiFields.add("proteinIds");
    openapiFields.add("transcriptIds");

    // a set of required properties/fields (JSON key names)
    openapiRequiredFields = new HashSet<String>();
  }

 /**
  * Validates the JSON Element and throws an exception if issues found
  *
  * @param jsonElement JSON Element
  * @throws IOException if the JSON Element is invalid with respect to EnsemblFilter
  */
  public static void validateJsonElement(JsonElement jsonElement) throws IOException {
      if (jsonElement == null) {
        if (!EnsemblFilter.openapiRequiredFields.isEmpty()) { // has required fields but JSON element is null
          throw new IllegalArgumentException(String.format("The required field(s) %s in EnsemblFilter is not found in the empty JSON string", EnsemblFilter.openapiRequiredFields.toString()));
        }
      }

      Set<Map.Entry<String, JsonElement>> entries = jsonElement.getAsJsonObject().entrySet();
      // check to see if the JSON string contains additional fields
      for (Map.Entry<String, JsonElement> entry : entries) {
        if (!EnsemblFilter.openapiFields.contains(entry.getKey())) {
          throw new IllegalArgumentException(String.format("The field `%s` in the JSON string is not defined in the `EnsemblFilter` properties. JSON: %s", entry.getKey(), jsonElement.toString()));
        }
      }
        JsonObject jsonObj = jsonElement.getAsJsonObject();
      // ensure the optional json data is an array if present
      if (jsonObj.get("geneIds") != null && !jsonObj.get("geneIds").isJsonNull() && !jsonObj.get("geneIds").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `geneIds` to be an array in the JSON string but got `%s`", jsonObj.get("geneIds").toString()));
      }
      // ensure the optional json data is an array if present
      if (jsonObj.get("hugoSymbols") != null && !jsonObj.get("hugoSymbols").isJsonNull() && !jsonObj.get("hugoSymbols").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `hugoSymbols` to be an array in the JSON string but got `%s`", jsonObj.get("hugoSymbols").toString()));
      }
      // ensure the optional json data is an array if present
      if (jsonObj.get("proteinIds") != null && !jsonObj.get("proteinIds").isJsonNull() && !jsonObj.get("proteinIds").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `proteinIds` to be an array in the JSON string but got `%s`", jsonObj.get("proteinIds").toString()));
      }
      // ensure the optional json data is an array if present
      if (jsonObj.get("transcriptIds") != null && !jsonObj.get("transcriptIds").isJsonNull() && !jsonObj.get("transcriptIds").isJsonArray()) {
        throw new IllegalArgumentException(String.format("Expected the field `transcriptIds` to be an array in the JSON string but got `%s`", jsonObj.get("transcriptIds").toString()));
      }
  }

  public static class CustomTypeAdapterFactory implements TypeAdapterFactory {
    @SuppressWarnings("unchecked")
    @Override
    public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
       if (!EnsemblFilter.class.isAssignableFrom(type.getRawType())) {
         return null; // this class only serializes 'EnsemblFilter' and its subtypes
       }
       final TypeAdapter<JsonElement> elementAdapter = gson.getAdapter(JsonElement.class);
       final TypeAdapter<EnsemblFilter> thisAdapter
                        = gson.getDelegateAdapter(this, TypeToken.get(EnsemblFilter.class));

       return (TypeAdapter<T>) new TypeAdapter<EnsemblFilter>() {
           @Override
           public void write(JsonWriter out, EnsemblFilter value) throws IOException {
             JsonObject obj = thisAdapter.toJsonTree(value).getAsJsonObject();
             elementAdapter.write(out, obj);
           }

           @Override
           public EnsemblFilter read(JsonReader in) throws IOException {
             JsonElement jsonElement = elementAdapter.read(in);
             validateJsonElement(jsonElement);
             return thisAdapter.fromJsonTree(jsonElement);
           }

       }.nullSafe();
    }
  }

 /**
  * Create an instance of EnsemblFilter given an JSON string
  *
  * @param jsonString JSON string
  * @return An instance of EnsemblFilter
  * @throws IOException if the JSON string is invalid with respect to EnsemblFilter
  */
  public static EnsemblFilter fromJson(String jsonString) throws IOException {
    return JSON.getGson().fromJson(jsonString, EnsemblFilter.class);
  }

 /**
  * Convert an instance of EnsemblFilter to an JSON string
  *
  * @return JSON string
  */
  public String toJson() {
    return JSON.getGson().toJson(this);
  }
}

