/*
 * Genome Nexus API
 * This page shows how to use HTTP requests to access the Genome Nexus API. There are more high level clients available in Python, R, JavaScript, TypeScript and various other languages as well as a command line client to annotate MAF and VCF. See https://docs.genomenexus.org/api.  Aside from programmatic clients there are web based tools to annotate variants, see https://docs.genomenexus.org/tools.   We currently only provide long-term support for the '/annotation' endpoint. The other endpoints might change.
 *
 * The version of the OpenAPI document: 2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package org.genome_nexus.client;

import org.genome_nexus.ApiException;
import org.genome_nexus.client.PfamDomain;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * API tests for PfamControllerApi
 */
@Disabled
public class PfamControllerApiTest {

    private final PfamControllerApi api = new PfamControllerApi();

    /**
     * Retrieves a PFAM domain by a PFAM domain ID
     *
     * @throws ApiException if the Api call fails
     */
    @Test
    public void fetchPfamDomainsByAccessionGETTest() throws ApiException {
        String pfamAccession = null;
        PfamDomain response = api.fetchPfamDomainsByAccessionGET(pfamAccession);
        // TODO: test validations
    }

    /**
     * Retrieves PFAM domains by PFAM domain accession IDs
     *
     * @throws ApiException if the Api call fails
     */
    @Test
    public void fetchPfamDomainsByPfamAccessionPOSTTest() throws ApiException {
        List<String> pfamAccessions = null;
        List<PfamDomain> response = api.fetchPfamDomainsByPfamAccessionPOST(pfamAccessions);
        // TODO: test validations
    }

}
